### DISCUSSION TITLE
Stream Deck doesn't recognize plugin

### DISCUSSION THREAD

**Discussion by wenkokke** [elgatosf/elgato#69] [Help] [answered]
I've created a basic plugin based on the CPP template, but my Stream Deck doesn't seem to recognize the plugin.

How do I debug what's going wrong?

Are there any unusual requirements on the stucture of the plugin folder name, other than a reverse DNS ending in `.sdPlugin`? For instance, is the `.is` TLD supported?

**Comment by sliptrixx** [comment, marked as answer]
If anyone's having issues with like this, here are a couple of things that helped fix my issue: 

- Ensure there's an `icon@2x.png` file
- Add at least one action to the list of actions in the manifest file

I was able to find these "issues" with my plugin by running the [packaging tool](https://docs.elgato.com/sdk/plugins/packaging), but I really wish that Elgato provided better error reporting functionality when debugging this.

EDIT: Just figured out that symlinks don't work for some reason. Copying to the folder works. 😡

**Reply by ChekTek** [reply]
This is a great answer. The [distribution tool](https://docs.elgato.com/sdk/plugins/packaging) is a great way to find issues with the plugin/manifest. We've also recently updated Stream Deck to be less strict with the contents of the manifest, which may help resolve some of these issues.

@sliptrixx symlinks should be working! Please make sure you are using the complete path. Sometimes symlink tools try to use the relative path to the original file, but that may not work with Stream Deck. If you are not able to get it working, please let me know your OS and Stream Deck version details and I will investigate further.

![CleanShot 2023-11-08 at 08 51 58](https://github.com/elgatosf/elgato/assets/6678919/3a75823c-35c2-4d2c-9d26-a09d7c0001d8)

**Reply by wenkokke** [reply]
For some reason, the Stream Deck refuses to load the plugin on MacOS if the file in CodePathWin is missing.

That may be another thing that'd be good to relax, for use in local testing.

**Reply by ChekTek** [reply]
We definitely should add some logging for this (I'll let the team know), but I believe this is happening because with `CodePathMac` Stream Deck then expects a `CodePathWin`. If you are working on a plugin that is only on one platform, or if the file is the same for both platforms, you'll want to use the `CodePath` property instead. That should resolve the issue.

**Reply by wenkokke** [reply]
> We definitely should add some logging for this (I'll let the team know), but I believe this is happening because with `CodePathMac` Stream Deck then expects a `CodePathWin`. If you are working on a plugin that is only on one platform, or if the file is the same for both platforms, you'll want to use the `CodePath` property instead. That should resolve the issue.

I can confirm it's not. Any permutation of setting CodePath and CodePathMac fails unless (1) I remove CodePathWin or (2) the .exe file for Windows exists.

**Reply by ChekTek** [reply]
Ah, my apologies, I misread your original problem. I thought the issue was with the property in the manifest itself, not the file being missing. I will relay this feedback to the Stream Deck team. 👍
